### DISCUSSION TITLE
yottadb.require where argument has multiple specifications

### DISCUSSION THREAD

**Discussion by AlainDsc** [anet-be/lua-yottadb#39] [Q&A] [answered]
Dear,

I wanted to use the function `yottadb.require` for call-in specifications, by giving as argument the specification itself. Like the documentation said: ' If the string contains : it is considered to be the call-in specification itself'.

However, my specification have 2 lines.

Following code seems to work:

```
ydb = require("yottadb")
local mexec = ydb.require("mexec: ydb_string_t*[1024] %X^%yadlua(I:ydb_string_t*)" ..
    "\013\010" .. "mexecp: ydb_string_t*[1024] %X^%yadlua(I:ydb_string_t*, I:ydb_string_t*)")
```

My question is : is this working by accident , or is this the right way to do so ?

**Comment by berwynhoyt** [comment, marked as answer]
Yes, that's a valid way to do it. However, a few stylistic notes:
* `"\013\10"` is more commonly known as `"\r\n"` in Lua
* The linux standard line ending is just a single `"\n"` not `"\r\n"`
* You don't need to end your quote and concatenate another one just to specify the line ending. So you can include your `\n` at the end of the text and don't need the extra `..` as follows:
``` lua
local mexec = ydb.require(
    "mexec: ydb_string_t*[1024] %X^%yadlua(I:ydb_string_t*)\n" ..
    "mexecp: ydb_string_t*[1024] %X^%yadlua(I:ydb_string_t*, I:ydb_string_t*)" )
```

or, using Lua long literals you can even do the following, which captures the `\n` that is inherent in the source:
``` lua
local mexec = ydb.require([[
    mexec: ydb_string_t*[1024] %X^%yadlua(I:ydb_string_t*)
    mexecp: ydb_string_t*[1024] %X^%yadlua(I:ydb_string_t*, I:ydb_string_t*)
]]
```

**Comment by AlainDsc** [comment]
Thank you, Berwyn, this is definitely an answer to my question.
I just added the '..' for clarity.
